src/partition/repart.c

```diff
@@ -215,10 +216,12 @@ DEFINE_PRIVATE_STRING_TABLE_LOOKUP_FROM_STRING_WITH_BOOLEAN(encrypt_mode, Encryp
 
 
 static uint64_t round_down_size(uint64_t v, uint64_t p) {
-        return (v / p) * p;
+        return p == 1 ? v : (v / p) * p;
```

Comment:
Isn't this actually making the code slower? A conditional might be more costly than an unnecessary operation.

I think this whole commit can be dropped. It's doing a microoptimization that either has no effect or might maybe even be a pessimization. The most important part is that it makes the code more complicated, and we don't want that.

Reply:
Yeah I've been thinking whether da83ab0bc8c4a675b44380283ed2e1a4b77df8a8 makes enough of a point, since devices with logical block size other than 512 and 4096 are probably scarce.